parser/digester.go

```diff
@@ -95,17 +95,17 @@ func Normalize(sql string) (result string) {
 // which removes general property of a statement but keeps specific property.
 //
 // for example: Normalize('select /*+ use_index(t, primary) */ 1 from b where a = 1') => 'select /*+ use_index(t, primary) */ ? from b where a = ?'
-func NormalizeKeepHint(sql string) (result string) {
+func NormalizeKeepHint(sql string, forBinding bool) (result string) {
 	d := digesterPool.Get().(*sqlDigester)
-	result = d.doNormalize(sql, true)
+	result = d.doNormalize(sql, true, forBinding)
 	digesterPool.Put(d)
 	return
 }
 
 // NormalizeDigest combines Normalize and DigestNormalized into one method.
-func NormalizeDigest(sql string) (normalized string, digest *Digest) {
+func NormalizeDigest(sql string, forBinding bool) (normalized string, digest *Digest) {
```

Comment:
How about keeping these exposed functions unchanged and creating a new exposed function? Otherwise this cause too many changes.
For example:
```
func  NormalizeKeepHint(sql string) (result string) {
    ...
    d.doNormalize(sql, true, false)
    ...
}

func NormalizeDigest(sql string) (normalized string, digest *Digest) {
    ...
    d.doNormalize(sql, true, false)
    ...
}

func NormalizeForBinding(sql string) {
    ...
    d.doNormalize(sql, false, true)
    ...
}
```

Reply:
Almost LGTM, the last one comment! @isabella0428